bigbluebutton-html5/imports/ui/components/video-provider/component.jsx

```diff
@@ -889,9 +891,51 @@ class VideoProvider extends Component {
     if (isAbleToAttach) {
       this.attach(peer, video);
       peer.attached = true;
+
+      if (isLocal) {
+        const deviceId = MediaStreamUtils.extractVideoDeviceId(peer.bbbVideoStream.mediaStream);
+        const { type, name } = getSessionVirtualBackgroundInfo(deviceId);
+
+        this.restoreVirtualBackground(peer.bbbVideoStream, type, name).catch(() => {
+          // if it fails, try again after 2 sec
+          setTimeout(() => {
```

Comment:
Why would a retry be needed?
The failure rate is supposed to be as rare as it can be, and in those scenarios I really doubt a retry would fix anything.

Reply:
Hmm! I have read somewhere that the virtual background is not guaranteed to start, so I did it that way. Do you think that this snippet of code is unnecessary?

Reply:
tl;dr yeah, I think the retry is unnecessary; catching + logging the error and notifying the user with a toast suffices.

> I have read somewhere that the virtual background is not guaranteed to start,
It's not guaranteed to start and that's why the service's start method throws. The thing is that if it throws once, it's hardly like that it would work on a retry (hence why it would be unnecessary). If it's failing intermittently, then it would be something we'd have to fix at the root rather than rely on an arbitrary retry.

Reply:
Got it!